unity/kuka-ar-unity/Assets/Project/Scripts/EventSystem/Controllers/Menu/TopMenuController.cs

```diff
@@ -0,0 +1,100 @@
+using System;
+using Project.Scripts.Connectivity.Models.KRLValues;
+using Project.Scripts.EventSystem.Enums;
+using Project.Scripts.EventSystem.Events;
+using Project.Scripts.TrackedRobots;
+using TMPro;
+using UnityEngine;
+
+namespace Project.Scripts.EventSystem.Controllers.Menu
+{
+    public class TopMenuController : MonoBehaviour
+    {
+        [Tooltip("Id of event controller")]
+        public int id;
+        
+        [Tooltip("Animation speed")]
+        public int transformFactor;
+        
+        [Tooltip("Menu drop height, counted from screen height [%]")]
+        public float dropScreenHeight;
+        
+        [Tooltip("Jogs component reference object")]
+        public GameObject jogs;
+
+        [Tooltip("Self reference to prevent null pointer exception on behavior")]
+        public GameObject topMenu;
```

Comment:
🤔 why no [Unity gameobject docs](https://docs.unity3d.com/ScriptReference/Component-gameObject.html)

Reply:
I'm gonna check this one, but if my memory is correct where was some problem with it.

Reply:
I modified this one, in dc5754206ceb4c513c8ec437408bef97fea77482 if you could check is it fine. There was small problem but i solved this somewhere during implementation of TOP MENU, so this suggestion works as it should now.